apps/settings/js/panels/full_developer_mode_final_warning/panel.js

```diff
@@ -0,0 +1,57 @@
+define(function(require) {
+  'use strict';
+
+  var SettingsService = require('modules/settings_service');
+  var DialogPanel = require('modules/dialog_panel');
+
+  return function() {
+    return DialogPanel({
+      onInit: function(panel) {
+        this._warningDialog = {
+          root: panel.querySelector('.warning-dialog'),
+          okBtn: panel.querySelector('.warning-dialog button[type="submit"]'),
+          cancelBtn:
+            panel.querySelector('.warning-dialog button[type="reset"]'),
+          warningInfo: panel.querySelector('.warning-dialog .warning-info')
+        };
+
+        this._count = 10;
+        this._updateWarningInfo(this._count);
+      },
+      onSubmit: function() {
+        if (this._count === 1) {
+          this._wipe();
+          return Promise.reject();
+        } else {
+          this._count--;
+          this._updateWarningInfo(this._count);
+          return Promise.reject();
+        }
+      },
+      onCancel: function() {
+        this._close();
+      },
+      _updateWarningInfo: function(count) {
+        window.navigator.mozL10n.setAttributes(this._warningDialog.warningInfo,
+          'enable-full-dev-mode-final-warning-msg', {
+            count: count
+          });
+      },
+      _close: function() {
+        SettingsService.navigate('home');
```

Comment:
Because it's not a good idea to call navgation API from a dialog, normally the dialog is a tool that used to ask for user's answer of something, for dialog itself, it should not realize the existence of the `panel`, to be honest, this should be done in caller side instead of callee.